testutil/test_helper.go

```diff
@@ -281,6 +281,28 @@ func (h *TestHelper) RetryFor(timeout time.Duration, fn func() error) error {
 	}
 }
 
+// HTTPGetCode sends a GET request to the given URL and returns the response code.
+// It retries requests for up to 30 seconds, giving pods time to start.
+func (h *TestHelper) HTTPGetCode(url string) (int, error) {
```

Comment:
Is this still needed for this PR? It looks except for the retries, the correct flags passed in to `curl` returns the response code.